### PR TITLE
Alert refactor

### PR DESCRIPTION
**Pull request by Duddino**
## Abstract

Add AlertController class, which stores alerts.
Add Alerts.vue and Alert.vue component, which display alerts. Alerts no longer spam

image

## Testing
- Open MPW and trigger some alerts
- Test that consecutive alerts correctly fold as one

### CODE REVIEW

**Review by JSKitty — APPROVED** [review]
**tACK** ff692c5e786fdc38926580426d90c61437328f2c

Looks great, works nicely for many alert types.

Console logs can be removed before merging.

**Review by panleone — APPROVED** [review]
tACK 8a53010d31ebdde16216ff640fa2cf53ac5cab46

The only concern I have is that `alerts` are never removed from the corresponding arrays, even when they expire... But I guess unless you leave MPW open for months it shouldn't create performance problems

File: scripts/alerts/Alerts.vue [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    const res = [];
+    let previousAlert;
+    let count = 1;
+    const pushAlert = () => {
+        if (previousAlert) {
+            const countStr = count === 1 ? '' : ` (x${count})`;
+            const timeout =
+                previousAlert.created + previousAlert.timeout - Date.now();
+            const show = timeout > 0;
+            if (!show) return;
```
[the remark this answers is not in the document]
Comment (Duddino) [reply scripts/alerts/Alerts.vue:18]: show is used in line 22 as well

**Comment by Duddino** [comment]
> The only concern I have is that `alerts` are never removed from the corresponding arrays, even when they expire... But I guess unless you leave MPW open for months it shouldn't create performance problems

This is to let us create a "see all alerts" page in the future. We cap the length of the alerts array at something like 1000 to prevent memory usage from growing uncontrolled

**Review by JSKitty — APPROVED** [review]
**tACK** 1415673e63b168e837d9bfffbca8727b2ef65a11

**Review by panleone — APPROVED** [review]
tACK ed63019018b53c4709866a9a0fafc4bd46e62422

**Pull request merged**